frontend/src/app/testQueue/TestCardForm/TestCardForm.utils.tsx

```diff
@@ -228,45 +227,140 @@ export const convertFromMultiplexResponse = (
   });
 };
 
-export const areAOEAnswersComplete = (
+export const AoeValidationErrorMessages = {
+  COMPLETE: "COMPLETE",
+  SYMPTOM_VALIDATION_ERROR: "SYMPTOM_VALIDATION_ERROR",
+  INCOMPLETE: "INCOMPLETE",
+  UNKNOWN: "UNKNOWN",
+} as const;
+
+const REQUIRED_AOE_QUESTIONS_BY_DISEASE: {
+  [key in AOEFormOption]: Array<keyof AoeQuestionResponses>;
+} = {
+  // AOE responses for COVID not required, but include in completion validation
+  // to show "are you sure" modal if not filled
+  [AOEFormOption.COVID]: ["pregnancy", "noSymptoms"],
+  [AOEFormOption.HIV]: ["pregnancy", "genderOfSexualPartners"],
+  [AOEFormOption.SYPHILIS]: [
+    "pregnancy",
+    "syphilisHistory",
+    "genderOfSexualPartners",
+    "noSymptoms",
+  ],
+  [AOEFormOption.NONE]: [],
+};
+
+export function generateAoeValidationState(
   formState: TestFormState,
   whichAOE: AOEFormOption
-) => {
-  if (whichAOE === AOEFormOption.COVID) {
-    const isPregnancyAnswered = !!formState.aoeResponses.pregnancy;
-    const hasNoSymptoms = formState.aoeResponses.noSymptoms;
-    if (formState.aoeResponses.noSymptoms === false) {
-      const symptoms = mapSymptomBoolLiteralsToBool(
-        formState.aoeResponses.symptoms,
-        respiratorySymptomDefinitions
-      );
-      const areSymptomsFilledIn = Object.values(symptoms).some((x) =>
-        x?.valueOf()
-      );
-      const isSymptomOnsetDateAnswered = !!formState.aoeResponses.symptomOnset;
-      return (
-        isPregnancyAnswered &&
-        !hasNoSymptoms &&
-        areSymptomsFilledIn &&
-        isSymptomOnsetDateAnswered
-      );
+): keyof typeof AoeValidationErrorMessages {
+  const {
+    allNonSymptomAoeQuestionsAnswered,
+    hasSymptomsQuestionAnswered,
+    symptomOnsetAndSelectionQuestionsAnswered,
+  } = validateAoeQuestions(formState, whichAOE);
+
+  if (allNonSymptomAoeQuestionsAnswered && hasSymptomsQuestionAnswered) {
+    if (symptomOnsetAndSelectionQuestionsAnswered) {
+      return AoeValidationErrorMessages.COMPLETE;
     }
-    return isPregnancyAnswered && hasNoSymptoms;
+    return AoeValidationErrorMessages.SYMPTOM_VALIDATION_ERROR;
+  } else if (
+    !allNonSymptomAoeQuestionsAnswered ||
+    !hasSymptomsQuestionAnswered
+  ) {
+    return AoeValidationErrorMessages.INCOMPLETE;
+  } else {
+    return AoeValidationErrorMessages.UNKNOWN;
   }
-  const hasPositiveHIVResult = formState.testResults.some(
-    (x) =>
-      x.diseaseName === MULTIPLEX_DISEASES.HIV &&
-      x.testResult === TEST_RESULTS.POSITIVE
-  );
-  if (whichAOE === AOEFormOption.HIV && hasPositiveHIVResult) {
-    const isPregnancyAnswered = !!formState.aoeResponses.pregnancy;
-    const isGenderOfSexualPartnersAnswered =
-      !!formState.aoeResponses.genderOfSexualPartners &&
-      formState.aoeResponses.genderOfSexualPartners.length > 0;
-    return isPregnancyAnswered && isGenderOfSexualPartnersAnswered;
+}
+
+export function validateAoeQuestions(
```

Comment:
See [this suggestion](https://github.com/CDCgov/prime-simplereport/pull/7713/files#r1607114012) about renaming